### PR TITLE
[Offload][NFC] Add offload subfolder and README

### PR DESCRIPTION
**Pull request by jdoerfert**
The readme only states the goal and has links to further information, e.g., our meetings.

### CODE REVIEW

File: offload/README.md
Comment (shiltian) [inline offload/README.md:13]:
```suggestion
discussions on [Discourse](https://discourse.llvm.org/tag/offload).
```

Comment (shiltian) [reply offload/README.md:13]: Unless you also opened a channel at Discord.

File: offload/README.md
Comment (shiltian) [inline offload/README.md:18]: This URL is the doc.
```suggestion
[invite.ics](https://drive.google.com/file/d/10Gv-MuEQ-VztJBqpQBggJclv2vU5VF0V/view?usp=sharing)
```

**Review by shiltian — APPROVED** [review]
LGTM. I'm not sure if we need special approval to add a folder at top level.

**Review by jplehr — APPROVED** [review]
LGTM

File: offload/README.md
Comment (saiislam) [inline offload/README.md:17]: Meeting time and ics file needs to be updated before landing this PR.

**Review by jhuber6 — APPROVED** [review]

**Pull request merged**
